modules/installation/proc-add-custom-app-config-file-ocp-operator.adoc

```diff
@@ -0,0 +1,95 @@
+
+.Procedure
+. From the *Developer* perspective, select the *ConfigMaps* tab.
+. Click *Create ConfigMap*.
+. Select the *YAML view* option in *Configure via* and use the following example as a base template to create a ConfigMap, such as `app-config-rhdh.yaml`:
++
+[source,yaml,subs="attributes+"]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: app-config-rhdh
+data:
+  "app-config-rhdh.yaml": |
+    app:
+      title: {product}
+      baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>
+    backend:
+      auth:
+        keys:
+          - secret: "${BACKEND_SECRET}"
+      baseUrl: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>
+      cors:
+        origin: https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>
+----
++
+.. Use the mandatory backend authentication key for {product} to reference an environment variable defined in an {ocp-short} Secret.
+.. Set the external URL of your {product} instance in the `app.baseUrl`, `backend.baseUrl` and `backend.cors.origin` fields of the application configuration. By default, the URL is similar to the following example:
+`pass:c[https://backstage-<CUSTOM_RESOURCE_NAME>-<NAMESPACE_NAME>.<OPENSHIFT_INGRESS_DOMAIN>;]`.
++
+[NOTE]
+====
+You can use the `oc get ingresses.config/cluster -o jsonpath='{.spec.domain}'` command to display your ingress domain. If you want to use a different host or sub-domain, customize the Custom Resource `spec.application.route` field and adjust the application configuration accordingly.
+====
+
+. Click *Create*.
+. Select the *Secrets* tab.
+. Click *Create Key/value Secret*.
+. Create a custom secret named `secrets-rhdh`.
+. Add a key named `BACKEND_SECRET` and a base64 encoded string as a value. Use a unique value for each {product} instance. For example, you can use the following command to generate a key from your terminal:
++
+[source,yaml]
+----
+node -p 'require("crypto").randomBytes(24).toString("base64")'
+----
+
+. Click *Create*.
+. Select the *Topology* view.
+. Click the overflow menu for the {product} instance that you want to use and select *Edit Backstage* to load the YAML view of the {product} instance.
++
+image::rhdh/operator-install-2.png[]
+
+. In the Custom Resource file, enter the name of the custom application configuration ConfigMap as the value for the `spec.application.appConfig.configMaps` field and the name of the custom Secret as the value for the `spec.application.extraEnvs.secrets` field. For example:
++
+[source, yaml]
+----
+spec:
+  application:
+    appConfig:
+      mountPath: /opt/app-root/src
+      configMaps:
+         - name: app-config-rhdh
+    extraEnvs:
+      secrets:
+         - name: secrets-rhdh
+    extraFiles:
+      mountPath: /opt/app-root/src
+    replicas: 1
+    route:
+      enabled: true
+  database:
+    enableLocalDb: true
+----
```

Comment:
In this YAML example, I would include the API version, kind, etc so that it's clear what the custom resource is that's being modified here. For example, if it was a config map:
```
apiVersion: v1
kind: ConfigMap
metadata:
  name: example
spec:....
```

I think it's good practice to always include this for context for users.